chore: upgrade packages

minasjam opened the pull request:
https://app.clickup.com/t/36057148/IK-2638

I upgraded all I thought reasonable, but I am as unsure as I can be, whether I did all I should have. Please have a look and let me know.

Also, Cypress works locally but not here :-/ It tells me that useNavigate is not exported from react-router-dom, but it should be with the version I have in. Wierd :-/ Anyone could point me to how to fix that?

dusan-kovacik-profiq commented on packages/web-ts/src/components/SetNewPassword.tsx:
```diff
@@ -10,7 +10,7 @@ const SetNewPassword = () => {
     IKUICore.renderSetNewPasswordForm({
       renderElementSelector: "#set-new-password-container",
-      token: referenceId,
+      token: referenceId || "",
```
Maybe it would be better to check whether the `referenceId` is defined before you call `IKUICore.renderSetNewPasswordForm`. Because in that case it's useless to call it.

dusan-kovacik-profiq commented on packages/web-ts/src/components/custom/CustomSetNewPassword.tsx:
```diff
@@ -9,7 +9,9 @@ const SetNewPassword = () => {
 
   const handleOnSubmit = React.useCallback(() => {
-    IKUIUserAPI.sendNewPassword(referenceId, newPassword).then(console.log).catch(console.log);
+    IKUIUserAPI.sendNewPassword(referenceId || "", newPassword)
```
Here the same. Check whether `referenceId` is defined before you call `IKUIUserAPI.sendNewPassword`.

dusan-kovacik-profiq commented on packages/web/package.json:
```diff
@@ -7,7 +7,7 @@
     "react-router": "^5.2.0",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3"
+    "react-scripts": "^5.0.1"
```
You forgot to update `react-router-dom` package. That's probably the reason why the cypress tests are failing.

minasjam replied:
updated up to web-ts version

minasjam commented:
@hanny21  @dusan-kovacik-profiq Per instructions I have updated packages in web-ts. 

I have also fixed cypress tests and updated the PR rules so that cypress checks web-ts in the pipeline. @kilimandzsaro please have a look at it.

hanny21 commented on packages/web-ts/package.json:
```diff
@@ -4,15 +4,15 @@
     "react-router": "5.2.0",
-    "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3",
+    "react-router-dom": "^6.3.0",
```
just wondering why we are not using the same version of `react-router` and `react-rouer-dom`? since `react-router-dom` contains everything what `router-router` does and has dependecy to it.

hanny21 commented on packages/web-ts/package.json:
```diff
@@ -4,15 +4,15 @@
   "dependencies": {
     "@indykiteone/jarvis-sdk-web": "0.1.17-alpha-8a65aa1",
-    "@types/node": "^12.0.0",
+    "@types/node": "17.0.23",
```
please put all types into dev dependencies together with typescript

hanny21 commented on packages/web-ts/.env-wonka:
```diff
@@ -0,0 +1,4 @@
+REACT_APP_BASE_URI=https://jarvis-dev.indykite.com
```
we should not commit any `.env` file.. if needed put it into `.gitignore`.. but we do not want to expose any credentials in public repo :wink:

hanny21 reviewed (changes requested):
left you few comments... also is there any reason why we cannot use latest react version, e.g. version 18? if so, would be nice to document it findings in the task you have

minasjam commented:
Hi @hanny21 @dusan-kovacik-profiq @kilimandzsaro , 
As the commit suggest the following has been done: 

- packages are updated
- cypress has been updated to v.10, tests have been fixed, dashboard has been added to PR, the pipeline now checks web-ts - big thanks to @kilimandzsaro for working on this and supporting me
- a workaround the double run of useeffect has been added

Please have a look, I have been at it for a bit too long, so I might have missed stuff.

kilimandzsaro reviewed (approved):
Good job! 👍

hanny21 commented on packages/web-ts/src/components/custom/CustomLogin.tsx:
```diff
@@ -17,44 +17,50 @@ interface IProps {
 
   const onSuccess = React.useCallback(
-    (data) => {
+    (data: any) => {
```
please do not use `any`, but assign correct type

hanny21 commented on packages/web-ts/src/components/custom/CustomRegistration.tsx:
```diff
@@ -27,11 +27,11 @@ const Registration: React.FC<IProps> = ({ setToken }) => {
 
   const onSuccess = React.useCallback(
-    (data) => {
+    (data: any) => {
```
please do not use `any`, but assign correct type

hanny21 commented on packages/web-ts/src/components/custom/CustomRegistration.tsx:
```diff
@@ -45,12 +45,22 @@ const Registration: React.FC<IProps> = ({ setToken }) => {
-      setRegisterOpts(response.opts);
-    })();
+    if (!ignore) {
+      console.log("Custom Registration");
```
remove

hanny21 reviewed (approved):
looks good :+1: , left just few comments..
I am also thinking about that double render problem... the workaround works, but look weird and I do not know what you think.. I am still undetermined which approach is the best (like I think one then after few hours I find argument agains)... now I am half day convinced that maybe that workaround is not sth we need ,because it will be just in development.. maybe we should just run the cypress tests against different environment? what do you think @kilimandzsaro ? I am not sure if it is just possible to set the node environment for the application running in cypress to something else that development (which might do the trick)

minasjam commented:
> looks good +1 , left just few comments.. I am also thinking about that double render problem... the workaround works, but look weird and I do not know what you think.. I am still undetermined which approach is the best (like I think one then after few hours I find argument agains)... now I am half day convinced that maybe that workaround is not sth we need ,because it will be just in development.. maybe we should just run the cypress tests against different environment? what do you think @kilimandzsaro ? I am not sure if it is just possible to set the node environment for the application running in cypress to something else that development (which might do the trick)

kilimandzsaro commented:
@hanny21 the tests runs against a local application what the test spins up. It's because it's using the latest code to test. It is easy to setup to run the test against a different environment, just need to setup the new baseUrl. Is this double rendering happens only when the app was spin up locally or it happens all the time? I think the double render by default is not nice and can be annoying. But, if we can determine an element which will be loaded the last on the page (and not part of the double render elements), then we can add a check to the tests to wait for that element. With this we maybe able to avoid the test failures. If there is such an element.

dusan-kovacik-profiq commented on packages/web-ts/src/App.tsx:
```diff
@@ -52,12 +52,17 @@ const App: React.FC = () => {
+
+    return () => {
+      ignore = true;
+    };
```
I think you don't need to care about the double rendering in this component as the `useEffect` calls the init function only and the function only sets some global variables.

minasjam replied:
We will look at the workaround with @hanny21 once more in Wednesday.. Hopefully we will get rid of it and set the Cypress tests to run not on dev :-)

minasjam replied:
Hi Dušan, 
So, the workaround has been reverted :-)

dusan-kovacik-profiq reviewed (approved):
Honestly, I don't like the workaround as it's on too many places - couldn't it be somehow moved just to one hook where the workaround would be applied? Anyway, we probably want to finish this task asap, so I'm giving you the approval, but we shouldn't keep the workaround in the current state :/

minasjam commented:
These last commits should fulfill the task [IK-3409](https://app.clickup.com/t/36057148/IK-3409) which will free me up to finish the original task.

hanny21 commented on packages/web-ts/.gitignore:
```diff
@@ -20,4 +20,4 @@ node_modules
 npm-debug.log*
 yarn-debug.log*
-yarn-error.log*
+yarn-error.log*
```
please fix :wink:

hanny21 reviewed (approved):
:+1:

hanny21 commented:
@minasjam also please can you rebase the branch? maybe too many commits... this is public repo, so make sure that git history looks good :wink:

dusan-kovacik-profiq commented on packages/web-ts/src/components/custom/CustomLogin.tsx:
```diff
@@ -50,11 +50,11 @@ const Login: React.FC<IProps> = ({ setToken }) => {
-    (data) => {
-      setToken(data);
-      history.push("/authenticated");
+    (data: DataTokenResponseType | Error) => {
```
`data` can also have an `Error` type?
